langchain/document_loaders/unstructured.py

```diff
@@ -65,6 +66,18 @@ def _get_elements(self) -> List:
     def _get_metadata(self) -> dict:
         """Get metadata."""
 
+    @abstractmethod
+    def _get_file_mod_date(self) -> str:
```

Comment:
Thanks for the advice given! I agree that it will be tricky designing it but I look forward to the challenge and hopefully can help you standardize metadata handling across the various loaders :)

I have taken the liberty of creating a script for this in a new folder `./langchain/document_loader/utils/`, called `metadata.py`. Basically, I wanted to abstract the metadata retrieval from the main document loader scripts and put them all together into `metadata.py`. Inside, we can have helper classes for each document loader to do the metadata retrieval.

For example, we can have the class `UnstructuredMetadata()` and it takes in the file path (as you have suggested) (or alternatively, file IO, if necessary), and its class methods will get metadata like "source", "create_at", "update_at", "mime_type" and "extension".